node/src/components/linear_chain.rs

```diff
@@ -352,9 +352,14 @@ where
                         block.hash(),
                         &fs.block_hash,
                         "block loaded from storage should have a matching block hash."
+                    );
+                    assert_eq!(
+                        block.header().era_id(),
+                        fs.era_id,
+                        "block loaded from storage should have a matching era_id."
```

Comment:
This is probably what you meant in the other comment?
Shouldn't be an assertion.

Reply:
Ah, I wasn't supposed to commit it :grimacing:

Reply:
Done in [`3bffb83` (#683)](https://github.com/CasperLabs/casper-node/pull/683/commits/3bffb831458819c2ed21e895cd3fcf945c06f1a4)